### PR TITLE
Remove hardcoded React Native configuration

### PR DESCRIPTION
**Pull request by grabbou**
Summary:
---------

Historically, React Native used to be tightly coupled with the CLI, support for Android and iOS was hardcoded. Since 0.60, we have made React Native ship with its own configuration, specifying Android and iOS platforms just like React Native Windows.

This PR removes "dead-code" that was only needed during the transition period. It wasn't used and called in the 0.60.

I have also removed some obsolete tests that were no longer needed.

### CODE REVIEW

File: packages/cli/src/tools/config/__tests__/index-test.ts [hunk 1]
```diff
@@ -79,8 +105,7 @@ test('should return dependencies from package.json', () => {
 
 test('should read a config of a dependency and use it to load other settings', () => {
   writeFiles(DIR, {
-    'node_modules/react-native/package.json': '{}',
-    'node_modules/react-native-test/package.json': '{}',
```
Comment (thymikee) [inline packages/cli/src/tools/config/__tests__/index-test.ts]: why this is removed?

Comment (grabbou) [reply packages/cli/src/tools/config/__tests__/index-test.ts]: Accident, but I am surprised that the tests are passing.

**Comment by grabbou** [comment]
Tests are failing on Windows. I will need a Windows machine to debug why the mocking logic doesn't work. 

If there's anybody with Windows that's willing to give it a go, let me know!

**Comment by thymikee** [comment]
@thecodrr would you like to help us debug this?

**Comment by thecodrr** [comment]
Yup. On it.

**Comment by thymikee** [comment]
@grabbou can you rebase and rerun CI btw?

**Comment by grabbou** [comment]
Any luck @thecodrr? We do happen to have same on #811. I have Windows machine with me as I am working on React Native Windows project right now. Will try to debug this in the coming days.

Meanwhile, help is appreciated :)

**Pull request merged**